bin.src/create_agn_db.py

```diff
@@ -122,84 +124,82 @@ def get_m_i(abs_mag_i, redshift):
         else:
             os.unlink(out_file_name)
 
-
     cat = GCRCatalogs.load_catalog(args.yaml_file)
 
-    cat_qties = cat.get_quantities(['redshift_true',
-                                    'blackHoleMass', 'blackHoleAccretionRate',
-                                    'galaxy_id'])
+    qty_list = cat.list_all_quantities(include_native=True)
+
+    use_direct_eddington = ('blackHoleEddingtonRatio' in qty_list)
 
-    valid = np.where(np.logical_and(cat_qties['blackHoleMass']>0.0,
-                                    cat_qties['blackHoleAccretionRate']>0.0))
+    if use_direct_eddington:
+        print('using native Eddington ratio')
+        cat_qties = cat.get_quantities(['redshift_true',
+                                        'blackHoleMass',
+                                        'blackHoleEddingtonRatio',
+                                        'galaxy_id', 'ra', 'dec'],
+                                        filters=[(lambda x: x>0.0,
+                                                  'blackHoleMass'),
+                                                 (lambda x: x>0.0,
+                                                  'blackHoleEddingtonRatio'),
+                                                 (lambda x:
+                                                     np.log10(x)>args.mbh_cut,
+                                                    'blackHoleMass')])
 
-    redshift = cat_qties['redshift_true'][valid]
-    bhm = cat_qties['blackHoleMass'][valid]
-    accretion_rate = cat_qties['blackHoleAccretionRate'][valid]
-    galaxy_id = cat_qties['galaxy_id'][valid]
+    else:
+        cat_qties = cat.get_quantities(['redshift_true',
+                                        'blackHoleMass',
+                                        'blackHoleAccretionRate',
+                                        'galaxy_id', 'ra', 'dec'],
+                                       filters=[(lambda x: x>0.0,
+                                                 'blackHoleMass'),
+                                                (lambda x: x>0.0,
+                                                 'blackHoleAccretionRate'),
+                                                (lambda x:
+                                                    np.log10(x)>args.mbh_cut,
+                                                    'blackHoleMass')])
```

Comment:
Not sure if you care, but Lines 135-145 are almost the same as Lines 148-158, except for changing `blackHoleEddingtonRatio` to `blackHoleAccretionRate`. Potentially can be refactored.

Reply:
Good point